q2_sourmash/_compute.py

```diff
@@ -5,40 +5,81 @@
-import qiime2.util
-import pandas as pd
-from q2_sourmash._format import MinHashSigJsonDirFormat
+import glob
 import os
 import subprocess
-import glob
 import sys
+from typing import Union
+
+import pandas as pd
+import qiime2.util
+from q2_types.per_sample_sequences import (
+    SingleLanePerSampleSingleEndFastqDirFmt
+)
+from q2_types_genomics.per_sample_data import MultiMAGSequencesDirFmt
+
+from q2_sourmash._format import MinHashSigJsonDirFormat
 
 
-def compute(sequence_file:SingleLanePerSampleSingleEndFastqDirFmt, ksizes: int, scaled: int, track_abundance: bool=True) -> MinHashSigJsonDirFormat:
+def _duplicate_mag_seqs(manifest, sequence_file):
+    output = MinHashSigJsonDirFormat()
+    for i, row in manifest.iterrows():
+        _, bin_id, src_fp = row["sample-id"], row["mag-id"], row["filename"]
+        src_fp = os.path.join(str(sequence_file), src_fp)
+        dest_fp = os.path.join(str(output), f'{bin_id}.fasta')
+        qiime2.util.duplicate(src_fp, dest_fp)
+    return output
 
-    #read in FastqManifestFormat to convert from sample name to filename
-    manifest = pd.read_csv(os.path.join(str(sequence_file), sequence_file.manifest.pathspec), header=0, comment='#')
 
+def _duplicate_seqs(manifest, sequence_file):
     output = MinHashSigJsonDirFormat()
+    for seq_file in glob.glob(os.path.join(str(sequence_file), '*.fastq.gz')):
+        src_fp = str(seq_file)
+        filename = os.path.basename(src_fp)
+        sample_id = list(
+            manifest[manifest['filename'] == filename]['sample-id']
+        )
+        dest_fp = os.path.join(str(output), f'{sample_id[0]}.fastq.gz')
+        qiime2.util.duplicate(src_fp, dest_fp)
+    return output
+
+
+def compute(
+        sequence_file: Union[
+            SingleLanePerSampleSingleEndFastqDirFmt, MultiMAGSequencesDirFmt
+        ],
+        ksizes: int,
+        scaled: int,
+        track_abundance: bool = True
+) -> MinHashSigJsonDirFormat:
+    # read in FastqManifestFormat to convert from sample name to filename
+    manifest = pd.read_csv(
+        os.path.join(str(sequence_file), sequence_file.manifest.pathspec),
+        header=0, comment='#'
+    )
 
-    for seq_file in glob.glob(os.path.join(str(sequence_file), '*fastq.gz')):
-        filepath = str(seq_file)
-        filename = os.path.basename(filepath)
-        sampleid = list(manifest[manifest['filename']==filename]['sample-id'])
-        # print(sampleid, filename)
-        qiime2.util.duplicate(filepath, os.path.join(str(output), sampleid[0]+'.fastq.gz'))
+    if isinstance(sequence_file, MultiMAGSequencesDirFmt):
+        output = _duplicate_mag_seqs(manifest, sequence_file)
+    else:
+        output = _duplicate_seqs(manifest, sequence_file)
 
-    command = ['sourmash', 'compute', str(output) + "/*", '--ksizes', str(ksizes), '--scaled', str(scaled)]
+    command = [
+        'sourmash', 'compute', f'{output}/*',
```

Comment:
in our next major release of sourmash, sourmash v5, we will be removing `sourmash compute` per https://github.com/sourmash-bio/sourmash/issues/1286.

It's been replaced with sourmash sketch; the following should work:
```
ksizestr = [ f"k={k}" for k in ksizes ]
ksizestr = ",".join(ksizestr)
abundstr = "noabund"
if track_abundance:
   abundstr = "abund"

command = ['sourmash', 'sketch', 'dna', f'{output}/*', '-p', f'scaled={scaled},{ksizestr},{abundstr}'
```

Reply:
Good to know, thanks for info! Should we then update it in a separate PR, once v5 is out?